language_service/src/lib.rs

```diff
@@ -145,4 +179,50 @@ impl<'a> LanguageService<'a> {
         trace!("get_hover result: {res:?}");
         res
     }
+
+    fn apply_configuration(&mut self, configuration: &WorkspaceConfigurationUpdate) -> bool {
+        let mut need_recompile = false;
+
+        if let Some(package_type) = configuration.package_type {
+            need_recompile |= self.configuration.package_type != package_type;
+            self.configuration.package_type = package_type;
+        }
+
+        if let Some(target_profile) = configuration.target_profile {
+            need_recompile |= self.configuration.target_profile != target_profile;
+            self.configuration.target_profile = target_profile;
+        }
+
+        trace!("need_recompile after configuration update: {need_recompile:?}");
+        need_recompile
+    }
+
+    fn recompile_all(&mut self) {
+        let mut sources = Vec::new();
+        // Gather the sources from the current document map
+        for (uri, state) in &self.document_map {
+            sources.push((
+                uri.clone(),
+                state.version,
+                state
+                    .compilation
+                    .unit
+                    .sources
+                    .find_by_name(uri)
+                    .expect("source should be found")
+                    .contents
+                    .clone(),
+            ));
+        }
+
+        // Now recompile everything with current settings
+        for source in sources {
```

Comment:
Are you not able to call update_document as you iterate over the document_map above? The interim sources vector seems unnecessary.

Also, does this also recompile the std lib for use with the new settings? With the `@Config` annotations there it behaves a little differently for base or full profile.

Reply:
Rust won't let me call `update_document` while holding another reference to the `document_map`, but I suppose I could refactor to use `iter_mut()` instead (and basically reuse the contents of the method `update_method` instead of calling it). There was some annoying code duplication when I did that, but I can give it another try.

Yes stdlib gets compiled here as well!